Pod/Classes/NSAttributedString+BonMotUtilities.m

```diff
@@ -0,0 +1,73 @@
+//
+//  NSAttributedString+BonMotUtilities.m
+//  BonMot
+//
+//  Created by Eliot Williams on 3/9/16.
+//
+//
+
+#import "NSAttributedString+BonMotUtilities.h"
+#import "BONChain.h"
+#import "BONSpecial.h"
+
+static NSString *const kUnassignedCharacterNamePrefix = @"\\N{<unassigned-";
+static NSString *const kUnassignedCharacterNameSuffix = @">}";
+
+@implementation NSAttributedString (BonMotUtilities)
+
+- (NSString *)bon_humanReadableString
+{
+    NSString *originalString = self.string;
+    NSMutableString *composedHumanReadableString = [NSMutableString string];
+
+    [originalString enumerateSubstringsInRange:NSMakeRange(0, originalString.length) options:NSStringEnumerationByComposedCharacterSequences usingBlock:^(NSString *BONCNullable substring, NSRange substringRange, NSRange enclosingRange, BOOL *BONCNonnull stop) {
+
+        static NSCharacterSet *s_newLineCharacterSet = nil;
+        if (!s_newLineCharacterSet) {
+            s_newLineCharacterSet = [NSCharacterSet newlineCharacterSet];
+        }
+        NSCharacterSet *s_whiteSpaceAndNewLinesSet = nil;
+        if (!s_whiteSpaceAndNewLinesSet) {
+            s_whiteSpaceAndNewLinesSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+        }
+
+        unichar character = [substring characterAtIndex:0];
+        NSString *specialCharacterSubstitutionString = [BONSpecial humanReadableStringDictionary][@(character)];
+
+        NSMutableString *mutableUnicodeName = substring.mutableCopy;
+        CFStringTransform((CFMutableStringRef)mutableUnicodeName, NULL, kCFStringTransformToUnicodeName, FALSE);
+
+        BONStringDict *attributes = [self attributesAtIndex:substringRange.location effectiveRange:NULL];
+        NSTextAttachment *attachment = attributes[NSAttachmentAttributeName];
+        UIImage *attachedImage;
+        attachedImage = attachment.image;
+
+        // Substitute attached images with @"{image#heightx#width}"
+        if (attachedImage) {
+            NSString *imageSubstitutionString = [NSString stringWithFormat:@"{image%.0fx%.0f}", attachedImage.size.height, attachedImage.size.width];
```

Comment:
#req I believe the image could be a fractional number of points. That gets tricky, because a dimension could best be represented as `36` or `36.5` or `36.6666667` depending on the device and image size/scale factor. Thinking it should probably use a fixed number of decimal places. Also thinking it should have a `- (NSString *)bon_humanReadableStringIncludingImageSize:(BOOL)includeImageSize`, in case you're writing a unit test that needs to work with different image sizes.